### ISSUE TITLE
Resources on dependencies limitations

### ISSUE THREAD

**Issue by maelle** [ropensci/ReLTER#71]
For info @oggioniale, these might be useful when reducing the number of Imports.

- https://recology.info/2018/10/limiting-dependencies/
- https://www.jimhester.com/talk/2019-rsc-deps/

**Comment by maelle** [comment]
What's the specific question? In any case yes it is better to not import (in the NAMESPACE) a whole package when you use only a few functions of that package.

**Comment by micha-silver** [comment]
Here's what I see (not sure if this is the correct way to test...)

```
direct_deps <- remotes::local_package_deps()
length(direct_deps)
[1] 46
all_deps <- renv::dependencies()
Finding R package dependencies ... Done!
length(all_deps$Package)
[1] 401
```
So we are declaring` @import` of 46 packages. And these are pulling along (recursively) another 350. And with all the `importFrom()` lines in NAMESPACE we are actually attaching all these right from the start. Is this a problem?

**Comment by maelle** [comment]
Yes it is a problem, especially as there is a hard CRAN limit on direct Imports in DESCRIPTION (20 if I remember correctly).
